### PR TITLE
added api key test button with resulsts

### PR DESCRIPTION
**Pull request by margud**

### CODE REVIEW

File: src/Service/ApiTestService.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     * @var ApiService
+     */
+    private $apiService;
+
+    /**
+     * @var Smarty
+     */
+    private $smarty;
+
+    public function __construct(Mollie $module, ApiService $apiService, Smarty $smarty)
```
Comment (tomas862) [inline src/Service/ApiTestService.php:53]: IMO Smarty or Twig is presentational parts of your page which should not be used in services. I would say, have a service which constructs template params and the template rendering itself leave it for controllers and hooks

File: src/Service/ApiTestService.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ * @link       https://www.mollie.nl
+ * @codingStandardsIgnoreStart
+ */
+
+namespace Mollie\Service;
+
+use Mollie;
+use Smarty;
+
+class ApiTestService
```
Comment (tomas862) [inline src/Service/ApiTestService.php:41]: if this class purpose is constructing the presentation layer , call it `ApiCredentialsBuilder`

always think about interfaces when creating services.

I would say that an interface like this:

```php
class TemplateBuilderInterface {

    public function getParams():
}
```

would be perfect

**Review by tomas862 — CHANGES_REQUESTED** [review]

File: controllers/admin/AdminMollieAjaxController.php [hunk 3]
```diff
@@ -18,6 +19,9 @@ public function postProcess()
             case 'resendPaymentMail':
```
Comment (aleksiuno) [inline controllers/admin/AdminMollieAjaxController.php:19]: Case statement argument strings could be added to the private or public constants (depends on if they will be used by other classes) for better readability and reusability. This is just an insight.

File: src/Service/ApiTestService.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            [
+                'testKeyInfo' => $testKeyInfo,
+                'liveKeyInfo' => $liveKeyInfo
+            ]
+        );
+
+        return $this->smarty->fetch($this->module->getLocalPath() . 'views/templates/admin/api_test_results.tpl');
+    }
+
+    public function getApiKeyInfo($apiKey)
```
Comment (aleksiuno) [inline src/Service/ApiTestService.php:75]: If the project is planned to run on PHP version 7.0 or higher, you can use argument and return types. If not it's a good idea to write a PHPDoc with argument and return types.

**Review by aleksiuno — APPROVED** [review]

**Review by aleksiuno — APPROVED** [review]

**Review by tomas862 — APPROVED** [review]

**Pull request merged**